src/path-watcher.js

```diff
@@ -608,83 +577,91 @@ class PathWatcherManager {
-  async createWatcher (rootPath, options, eventCallback) {
-    if (this.isShuttingDown) {
-      await this.constructor.transitionPromise
-      return PathWatcherManager.active().createWatcher(rootPath, options, eventCallback)
-    }
-
+  async watchPath (rootPath, options, eventCallback) {
     if (this.useExperimentalWatcher()) {
-      if (this.setting === 'poll') {
-        options.poll = true
+      if (!this.notifyWatcher) {
+        const options = {
+          onError: (error) => {
+            throw new Error(`Error watching file system: ${error}`)
+          }
+        }
+        if (this.setting === 'poll') {
+          options.pollInterval = this.pollInterval
+        }
+        this.notifyWatcher = new NotifyWatcher(options)
       }
 
-      const w = await watcher.watchPath(rootPath, options, eventCallback)
-      this.live.set(rootPath, w.native)
+      const watch = await this.notifyWatcher.watchPath(rootPath, event => {
+        if (event.action === 'error') {
+          watch.emitter.emit('error', event.description)
+          throw new Error(`Error watching file system at "${event.path}": ${event.description}`)
+        } else {
+          eventCallback(event)
+        }
+      })
+      watch.emitter = new Emitter()
+      watch.onDidError = function (handler) {
+        return this.emitter.on('error', handler)
+      }
+      return watch
+    } else {
+      const w = new PathWatcher(this.nativeRegistry, rootPath, options)
+      w.onDidChange(eventCallback)
+      await w.getStartPromise()
       return w
     }
-
-    const w = new PathWatcher(this.nativeRegistry, rootPath, options)
-    w.onDidChange(eventCallback)
-    await w.getStartPromise()
-    return w
-  }
-
-  // Private: Directly access the {NativeWatcherRegistry}.
-  getRegistry () {
-    if (this.useExperimentalWatcher()) {
-      return watcher.getRegistry()
-    }
-
-    return this.nativeRegistry
-  }
-
-  // Private: Sample watcher usage statistics. Only available for experimental watchers.
-  status () {
-    if (this.useExperimentalWatcher()) {
-      return watcher.status()
-    }
-
-    return {}
-  }
-
-  // Private: Return a {String} depicting the currently active native watchers.
-  print () {
-    if (this.useExperimentalWatcher()) {
-      return watcher.printWatchers()
-    }
-
-    return this.nativeRegistry.print()
   }
 
   // Private: Stop all living watchers.
   //
   // Returns a {Promise} that resolves when all native watcher resources are disposed.
-  stopAllWatchers () {
+  async stopAllWatchers () {
     if (this.useExperimentalWatcher()) {
-      return watcher.stopAllWatchers()
+      await this.notifyWatcher.kill()
+      this.notifyWatcher = null
+    } else {
+      await Promise.all(
+        Array.from(this.live, ([, w]) => w.stop())
+      )
     }
-
-    return Promise.all(
-      Array.from(this.live, ([, w]) => w.stop())
-    )
   }
+
```

Comment:
@smashwilson I left these methods in here to get your thoughts on whether I can go ahead and delete them. They seem to be unused.

Reply:
👍 Yeah, these are diagnostic methods used to investigate stuff like failing tests in CI, or as a diagnostic tool to investigate odd behaviors. If they don't make sense with @atom/notify, :fire: